<?xml version="1.0"?>
<robot name="basic_robot">
    <link name="chassis">
        <visual>
            <geometry>
                <box size="0.825 0.825 0.075" />
            </geometry>
            <origin xyz="0 0 0" />
            <material name="white">
                <color rgba="1.0 1.0 1.0 1.0"/>
            </material>
        </visual>
    </link>
    <link name="left_wheel">
        <visual>
            <geometry>
                <cylinder length="0.05" radius="0.4" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0.025" />
            <material name="blue">
                <color rgba="0.0 0.0 1.0 1.0"/>
            </material>
        </visual>
    </link>
    <link name="right_wheel">
        <visual>
            <geometry>
                <cylinder length="0.05" radius="0.4" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 -0.025" />
            <material name="red">
                <color rgba="1.0 0.0 0.0 1.0"/>
            </material>
        </visual>
    </link>
    <link name="lidar_dome">
        <visual>
            <geometry>
                <sphere radius="0.03125" />
            </geometry>
            <material name="green">
                <color rgba="0.0 1.0 0.0 1.0"/>
            </material>
        </visual>
    </link>
    <joint name="chassis_to_left_wheel" type="continuous">
        <parent link="chassis" />
        <child link="left_wheel" />
        <origin xyz="0 0.4125 0" rpy="-1.57079632679 0 0" />
        <axis xyz="0 0 1" />
    </joint>
    <joint name="chassis_to_right_wheel" type="continuous">
        <parent link="chassis" />
        <child link="right_wheel" />
        <origin xyz="0 -0.4125 0" rpy="-1.57079632679 0 0" />
        <axis xyz="0 0 1" />
    </joint>
    <joint name="chassis_to_dome" type="fixed">
        <parent link="chassis" />
        <child link="lidar_dome" />
        <origin xyz="0.3 0 0.03125" />
    </joint>
</robot>